src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include <UIAutomationCore.h>
+#include "TermControlAutomationPeer.h"
+#include "TermControl.h"
+#include "TermControlAutomationPeer.g.cpp"
+
+#include "XamlUiaTextRange.h"
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    TermControlAutomationPeer::TermControlAutomationPeer(winrt::Microsoft::Terminal::TerminalControl::implementation::TermControl const& owner) :
+        TermControlAutomationPeerT<TermControlAutomationPeer>(owner),
+        _uiaProvider{ owner.GetRenderData(), nullptr, std::bind(&TermControlAutomationPeer::GetBoundingRectWrapped, this) } {};
+
+    winrt::hstring TermControlAutomationPeer::GetClassNameCore() const
+    {
+        return L"TerminalControl";
+    }
+
+    AutomationControlType TermControlAutomationPeer::GetAutomationControlTypeCore() const
+    {
+        return AutomationControlType::Text;
+    }
+
+    winrt::hstring TermControlAutomationPeer::GetLocalizedControlTypeCore() const
+    {
+        return L"TerminalControl";

```

Comment:
TODO this must be localized!

Reply:
Would this fall under #1484 ?

Reply:
We should be making efforts to put things into resource files when authoring new ones even if there's a task to cleanup.

That is, we shouldn't be actively making a bigger mess just because there's a task for it to be cleaned.